packages/server/lib/gui/files.ts

```diff
@@ -2,40 +2,30 @@ import openProject from '../open_project'
 import { createFile } from '../util/spec_writer'
 import { showSaveDialog } from './dialog'
 
-export const showDialogAndCreateSpec = () => {
-  return openProject.getConfig()
-  .then((cfg) => {
-    return showSaveDialog(cfg.integrationFolder).then((path) => {
-      return {
-        cfg,
-        path,
-      }
-    })
-  })
-  .tap(({ path }) => {
-    // only create file if they selected a file
-    if (path) {
-      return createFile(path)
-    }
+export const showDialogAndCreateSpec = async () => {
+  const cfg = openProject.getConfig()
+
+  const path = await showSaveDialog(cfg.integrationFolder)
 
-    return
-  })
-  .then(({ cfg, path }) => {
-    if (!path) {
-      return {
-        specs: null,
-        path,
-      }
+  if (!path) {
+    return {
+      specs: null,
+      path,
     }
+  }
+
+  // only create file if they selected a file
+  if (path) {
+    createFile(path)
```

Comment:
`createFile` returns a promise, so this should be awaited, right?

Reply:
Right, good catch - I forgot that `fs.writeFile` is actually wrapped with promisify.